### PR TITLE
limiting the number of excursion in an iterator

### PR DESCRIPTION
**Pull request by mducoffe**

### CODE REVIEW

**Comment by dwf** [comment]
Can you provide an example of where you might use this? It seems like most schemes provide a way to limit the number of returned batches/examples automatically based on their constructor arguments.

**Comment by mducoffe** [comment]
When evaluating on the validation set I would like to use a random subset
of the minibatches using this iterator and ShuffledScheme. Like that I can
have a rough idea of the performance when trying the hyperparameters

2015-06-22 16:12 GMT-04:00 David Warde-Farley <notifications@github.com>:

**Comment by dwf** [comment]
Okay. The name could probably be better too -- it's unclear in what way it would be "limited". Perhaps "TruncatedEpochScheme"? @rizar might have an opinion.

**Comment by vdumoulin** [comment]
I'd be okay with `TruncatedEpochScheme`.

File: fuel/schemes.py [hunk 1]
```diff
@@ -232,3 +232,17 @@ def get_request_iterator(self):
         indices = list(self.indices)
         self.rng.shuffle(indices)
         return iter_(indices)
+
+@add_metaclass(ABCMeta)
+class LimitedScheme(IterationScheme):
+    """Iterate over the sequence given an iterationScheme object
+        but in a limited number of excursions defined by times
+    """
+    def __init__(self, iteration_scheme, times, *args, **kwargs):
```
Comment (vdumoulin) [inline fuel/schemes.py]: You can safely remove `*args, **kwargs`. It's used in other iterators (e.g. `ShuffledExampleScheme`) to pass arbitrary args and kwargs to the superclass' `__init__` method, but in this case you're not calling any other `__init__` method.

File: fuel/schemes.py [hunk 2]
```diff
@@ -232,3 +232,17 @@ def get_request_iterator(self):
...
+        self.iteration_scheme = iteration_scheme
+        self.times = times
+        assert self.times >=1, "Error : you need a positiv number of mini batches"
+
+    def get_request_iterator(self):
+        it = self.iteration_scheme.get_request_iterator()
+        return iter_([it.next() for n in range(self.times)])
```
Comment (vdumoulin) [inline fuel/schemes.py]: Is there a reason why you're coercing `it.next() for n in range(self.times)` into a list?

Comment (dwf) [inline fuel/schemes.py]: @vdumoulin You actually would have to do that for the result to be
picklable (generators are not). But you're better off just using
`picklable_itertools.islice`.

Comment (vdumoulin) [inline fuel/schemes.py]: TIL, then!

**Comment by vdumoulin** [comment]
@mducoffe I'm doing a cleanup of the open PRs. Would you still like to go ahead with this one?
